### ISSUE TITLE
convert to tensorRT model

### ISSUE THREAD

**Issue by shuaizzZ** [Res2Net/Res2Net-PretrainedModels#33]
Assertion failed: *tensor = importer_ctx->network()->addInput( input.name().c_str(), trt_dtype, trt_dims)

**Comment by gasvn** [comment]
Maybe you can send me the converting codes so I can check it for you.

**Comment by shuaizzZ** [comment]
I made a mistake and I have solved this problem.

**Comment by shuaizzZ** [comment]
It seems that using nn.ModuleList() takes up more GPU memory than explicitly defining each module.

**Comment by shuaizzZ** [comment]
Is it because of this reason that res2net takes up more GPU memory?

**Comment by ishgao** [comment]
Yes, this may be part of reason that res2net uses more memory. Another part is that the concatation in pytorch is not optimized well.

**Comment by gasvn** [comment]
Yes, this may be part of reason that res2net uses more memory. Another part is that the concatation in pytorch is not optimized well.

**Comment by shuaizzZ** [comment]
When I tried to apply the idea of res2net to other networks, I found that the time consumption did not increase, but it used significantly more memory.

**Comment by shuaizzZ** [comment]
If this problem can be improved, it is really great.

**Comment by gasvn** [comment]
If you don't need to finetune the batchnorm layer, merge the batchnorm layer into the conv layer will save you lots of memory.
Also, use half-precision will cost you only half of the memory and almost have no performance drop.
Let's hope that pytorch will optimize the concat op. 
And I will let you know if I find another solution to this issue.

**Comment by gasvn** [comment]
In experience, the extra memory used by res2net is affordable. Maybe you can check if you use the in-place version of relu. In-place can also save lots of memory.

**Comment by shuaizzZ** [comment]
I tried the following changes to the cat method, and found that res2net uses almost the same memory as resnet:
==>Code before change
out = sp
...
out = torch.cat((out, sp), 1)
==>Code after change
out = [sp]
...
out.append(sp)
out = torch.cat(out, 1)

**Comment by shuaizzZ** [comment]
I am verifying that there is no problem, what do you think?

**Comment by ishgao** [comment]
Thanks! I think your version is quite well. I am gonna test it and update the code.

**Comment by gasvn** [comment]
Thanks! I think your version is quite well. I am gonna test it and update the code.

**Comment by shuaizzZ** [comment]
Haha, but I haven't figured out why this works.

**Comment by gasvn** [comment]
I have modified the code. And it's strange that the new version have almost no memory change during inference.
Here is my code:
https://gist.github.com/gasvn/0c359e2d6a79760196c828f763879273

I am using pytoch1.3. Maybe it's about the pytorch version.
Or this modification saves the memory during training?

**Comment by shuaizzZ** [comment]
I am also using pytoch1.3. 
Sorry, I tried this change on my network first, and it did save memory during training and inference. Then I tested on res2net, but I forgot to switch the network during the test, so the test result is wrong. I just re-tested it. In fact, it really did not save memory, and I also felt very strange.

**Comment by gasvn** [comment]
Thanks anyway. Fell free to let me know if you have other problems.
